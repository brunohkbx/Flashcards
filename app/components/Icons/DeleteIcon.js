import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

const DeleteIcon = ({ size, color, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <MaterialCommunityIcons
        name="delete"
        color={color}
        size={size}
      />
    </TouchableOpacity>
  );
};

DeleteIcon.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
  onPress: PropTypes.func.isRequired
};

export default DeleteIcon;
