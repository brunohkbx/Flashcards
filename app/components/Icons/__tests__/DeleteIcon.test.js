import React from 'react';
import { shallow } from 'enzyme';
import DeleteIcon from '../DeleteIcon';

describe('DeleteIcon', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      onPress: jest.fn()
    }, propOverrides);

    const wrapper = shallow(<DeleteIcon {...defaultProps} />);

    const press = () => wrapper.simulate('press');

    return { wrapper, press };
  };

  it('renders properly', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });


  test('onPress calls onPress', () => {
    const mockOnPress = jest.fn();
    const { press } = setup({ onPress: mockOnPress });

    press();

    expect(mockOnPress).toHaveBeenCalled();
  });
});
