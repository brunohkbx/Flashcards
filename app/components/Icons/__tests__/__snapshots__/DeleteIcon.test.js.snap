// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteIcon renders properly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
>
  <Icon
    allowFontScaling={false}
    name="delete"
    size={12}
  />
</TouchableOpacity>
`;
